test/lib-interaction/Makefile

```diff
@@ -74,20 +67,20 @@ $(OutFiles) : %.out : $(wildcard %.agda %.lagda) $(wildcard %.in %.in_ghci)
 
 # Comparing output
 $(Tests) : %.cmp : %.out
-	@-mkdir $(TMPDIR)
+	@-mkdir -p $(TMPDIR)
 	@$(setup_$*)
 	@rm -f "$(@:.cmp=.agdai)"
 	@echo $*
 	@$(run_test) > $*.tmp
-	@if diff -b $*.out $*.tmp; \
+	@if diff -q -b $*.out $*.tmp; \
 		then rm -f $*.tmp; true; \
 		else \
 			echo "=== Old output ==="; \
 			cat $*.out; \
 			echo "=== New output ==="; \
 			cat $*.tmp; \
 			echo "=== Diff ==="; \
-			diff -b $*.out $*.tmp; \
+			( git diff --no-index --no-ext-diff -w $*.out $*.tmp || true ) ; \
```

Comment:
Are you looking for the `--exit-code` option to `git diff`?

Reply:
No, it was a conscious choice, though possibly deserved a comment. Pagers can sometimes propagate exit codes unrelated to the diff status. Since this is specifically for interactive use it's likely a pager or other user-configured filter is involved. For example, if you use `pygmentize`, for a long time that used to exit from an `EPIPE` when `less` ends. Some other pagers and filters can exit with status if you decline to view the whole output. So this status suppression is intentionally more heavy-handed than `--no-exit-code` permits.

(If you're referring to the `--no-ext-diff`, I'm attempting to keep it terminal-based due to the interactive prompt; some people have GUI diff drivers).